src/inputs.ts

```diff
@@ -69,19 +63,32 @@ export function getInputs(): Inputs {
     format: core.getBooleanInput("format"),
     breaking: core.getBooleanInput("breaking"),
     breaking_against: core.getInput("breaking_against"),
-    breaking_against_config: core.getInput("breaking_against_config"),
-    breaking_limit_to_input_files: core.getBooleanInput(
-      "breaking_limit_to_input_files",
-    ),
     push: core.getBooleanInput("push"),
-    push_create: core.getBooleanInput("push_create"),
-    push_create_visibility: core.getInput("push_create_visibility"),
-    push_labels: core.getMultilineInput("push_labels"),
-    push_git_metadata: core.getBooleanInput("push_git_metadata"),
-    push_source_control_url: core.getInput("push_source_control_url"),
+    push_disable_create: core.getBooleanInput("push_disable_create"),
     archive: core.getBooleanInput("archive"),
-    archive_labels: core.getMultilineInput("archive_labels"),
+    archive_labels: [],
   };
+  if (github.context.eventName === "push") {
+    const event = github.context.payload as PushEvent;
+    core.info(`The head commit is: ${event.before}`);
+    if (inputs.breaking_against === "") {
+      inputs.breaking_against = `${event.repository.clone_url}#format=git,commit=${event.before}`;
```

Comment:
so is this defaulting breaking_against to the previous commit in the branch for a push?

Reply:
Yes, for push events if someone enables `breaking: true` we use the previous commit. Same as the behaviour before where we first try for pull request then default to the before sha:
```
default: ${{ github.event.repository.clone_url }}#format=git,commit=${{ github.event.pull_request.base.sha || github.event.before }}
```
But we now set the subdir (I just added this) to avoid users having to configure breaking against for `input` changes. If the user does set the `input` to a non directory input they will get an error showing the breaking against value is invalid from `buf`.